### ISSUE TITLE
New button for external url

### ISSUE THREAD

**Issue by duylong** [statusengine/interface#44]
Hi,

What do you think of creating a new button in host's details with a drop-down list to external URLs that we would have defined in the conf?

```
external_urls: 1
external_url_lists:
  Thruk: https://www.example.com/thruk/#cgi-bin/extinfo.cgi?type=1&host={{HOSTNAME}}
  Other: ...
```

**Comment by nook24** [comment]
Unfortunately most of the configuration is unknown to the Statusengine Interface, like IP addresses, contact configuration, check commands and so on.
Unlike [Statusengine 2](https://github.com/nook24/statusengine) which is a drop-in replacement for NDOUtils and has access to all of the Naemon configuration, Statusengine 3 was completely designed as a configuration less but fast status only database.

This is also the reason why it is more or less not possible to show the ip address of a host like you suggested in #41.

The idea is to have a separate configuration database that is used to generate the Naemon configuration and query Statusengin to get the current state of an object.

The question is, is it helpful for you to have one or more URLs for all hosts and services, where only hostname, service_description and status related information can be replaced dynamically?
If yes, this is not a big deal to implement.

But it will not be as dynamic as you may expect due to the lack of information.


With https://github.com/it-novum/openITCOCKPIT (the project of my employer) you can join the Statusengine database with a configuration database to get features like dynamic urls and many more:
![Bildschirmfoto 2020-05-19 um 20 17 35](https://user-images.githubusercontent.com/9019992/82365288-c09eac80-9a10-11ea-8e5c-68032834f755.png)

Maybe this could be interesting for you or maybe not

**Comment by duylong** [comment]
I had seen the openITCOCKPIT project but I didn't want to add another solution to my architecture. I want to keep it simple :)

Yes, having the hostname or service_description are sufficient, there is no need to have further in Statusengine 3. The goal is to have an identifier (common to all other external solutions), we don't need a persistent database.

Statusengine stores logs/metrics in a MySQL database, can't we enrich a Redis database to use it as a data cache for example?

**Comment by nook24** [comment]
Done :)

![Bildschirmfoto 2021-02-27 um 14 09 27](https://user-images.githubusercontent.com/9019992/109388482-a121a200-7907-11eb-87c4-4fd7001fad6f.png)


https://github.com/statusengine/interface/blob/89a23f59f750f20b3ac0fd0b9b5cadc3cce8cef8/etc/config.yml.example#L231-L237

**Comment by nook24** [comment]
Implemented and released with version 3.7
